core/types/transaction_signing.go

```diff
@@ -194,14 +205,19 @@ func (s eip2930Signer) Sender(tx *Transaction) (common.Address, error) {
 }
 
 func (s eip2930Signer) SignatureValues(tx *Transaction, sig []byte) (R, S, V *big.Int, err error) {
-	switch tx.Type() {
-	case LegacyTxType:
+	switch txdata := tx.inner.(type) {
+	case *LegacyTx:
 		R, S, V = decodeSignature(sig)
 		if s.chainId.Sign() != 0 {
```

Comment:
What if the legacy `chainId` is `nil`, in the case of a homestead-signed tx. Don't we need to check for that?

Reply:
`s.chainId` is always non-nil.